facebook/facebook-json.c

```diff
@@ -60,8 +62,15 @@ json_value *fb_json_new(const gchar *data, gsize length, GError **err)
         return json;
     }
 
+    /* Ensure it's null-terminated before passing it to g_strescape() */
+    dstr = g_strndup(data, MIN(length, 400));
```

Comment:
Arbitrary number, but the irc character limit is 512, minus len(":root!root@localhost PRIVMSG &bitlbee :JSON len=1234: ") which is 54 chars and the nick / hostname / length portions are variable length. So just picked 400 arbitrarily to give it some margin.